hazelcast/src/main/java/com/hazelcast/client/map/impl/nearcache/NearCachedClientMapProxy.java

```diff
@@ -257,6 +257,18 @@ protected InternalCompletableFuture<V> removeAsyncInternal(Object key) {
         return future;
     }
 
+    @Override
+    protected InternalCompletableFuture<Void> deleteAsyncInternal(Object key) {
+        key = toNearCacheKey(key);
+        InternalCompletableFuture<Void> future;
+        try {
+            future = super.deleteAsyncInternal(key);
+        } finally {
+            invalidateNearCache(key);
```

Comment:
I'm not sure if this works as expected. The problem is that the call is made async, so the near cache is invalidated before the call has been completed.

Reply:
Note: it is the same for other methods, say putAsyncInternal

Reply:
I don't know why we don't call invalidateNearCache in a whenComplete() call or something like that.

Reply:
I see that the same pattern is used both in the client and server-side proxies for all invalidating async methods. Since we are invalidating, but not adding anything to the local cache, I don't think it is a problem. At worst, we will remove the key from the local cache, but it won't be removed in the remote if the async call fails. I believe this pattern is fine, as long as we don't add anything to the cache. @ahmetmircik WDYT?

Reply:
@mdumandag Agree that invalidation should be done inside callback after we know that operation is completed; this is the ideal case and can prevent unneeded lookups to remote nodes. An improvement issue can be opened for this. 
BTW we do invalidations in async callbacks for ICache near-cache in similar situations. Aligning  both behaviors will be good also.